Core/include/Acts/Seeding/SeedFinderUtils.ipp

```diff
@@ -11,21 +11,37 @@ template <typename external_spacepoint_t>
 LinCircle transformCoordinates(
     const InternalSpacePoint<external_spacepoint_t>& sp,
     const InternalSpacePoint<external_spacepoint_t>& spM, bool bottom) {
+  auto extractFunction =
+      [](const InternalSpacePoint<external_spacepoint_t>& obj)
+      -> std::array<float, 6> {
+    std::array<float, 6> output{obj.x(),      obj.y(),         obj.z(),
+                                obj.radius(), obj.varianceR(), obj.varianceZ()};
+    return output;
+  };
+
+  return transformCoordinates<InternalSpacePoint<external_spacepoint_t>>(
+      sp, spM, bottom, extractFunction);
+}
+
+template <typename external_spacepoint_t>
+LinCircle transformCoordinates(
+    const external_spacepoint_t& sp, const external_spacepoint_t& spM,
+    bool bottom,
+    std::function<std::array<float, 6>(const external_spacepoint_t&)>
+        extractFunction) {
   // The computation inside this function is exactly identical to that in the
   // vectorized version of this function, except that it operates on a single
   // spacepoint. Please see the other version of this function for more
   // detailed comments.
-  float xM = spM.x();
-  float yM = spM.y();
-  float zM = spM.z();
-  float rM = spM.radius();
-  float varianceZM = spM.varianceZ();
-  float varianceRM = spM.varianceR();
+
```

Comment:
If you replace `std::function`, you could consider using SFINAE to add a `static_assert` here that gives a helpful error message.

Reply:
I should have addressed this now